addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -314,6 +314,16 @@ MockServer.include({
         }
         return mockWriteResult;
     },
+    /**
+     * @override
+     */
+    _writeRecord(model, values, id) {
```

Comment:
Solution was to overwrite `_writeRecord` to have this behaviour in both `pyEnv` and `this.data` the safest way possible.

The adapt of `im_livechat` above could be reverted. But we want sooner or later to adapt, so up to reviewer whether it's fine or prefer revert.